src/java.base/share/classes/java/text/ChoiceFormat.java

```diff
@@ -558,7 +558,23 @@ StringBuf format(double number, StringBuf toAppendTo,
     }
 
     /**
-     * Parses a Number from the input text.
+     * Parses a {@code Number} from the input text. The value returned is the
+     * {@code limit} corresponding to the {@code format} that is the longest
+     * substring of the input text. Matching is done in ascending order, when
+     * multiple {@code formats} match the text equivalently in strength, the
+     * first matching {@code limit} is returned. If there is no match, {@code
+     * Double.NaN} is returned.
+     * <p>
+     * For example,
+     * {@snippet lang=java :
+     * var fmt = new ChoiceFormat("0#foo|1#bar|2#baz");
+     * fmt.parse("baz", new ParsePosition(0)); // returns 2
+     * fmt.parse("quux", new ParsePosition(0)); // returns NaN
+     * }
+     *
+     * @implNote The {@code Number} subtype returned by the JDK reference
+     * implementation of this method is always {@code Double}.
```

Comment:
Do we need to use `@implNote` here? Since choices are `double`s (as in the class description), I think we can safely say this returns a `Double` as in normative text. If some implementation returns an `Integer`, I think it is a bug. Returning a `Double.NaN` for no-match may be considered implNote though (one might throw an exception).

Reply:
I was either way on the `implNote`, since I thought an implementation could decide to normalize a double limit to an integral type. However that's probably unlikely and I agree the wording can be fine as normative since ChoiceFormat is composed of doubles.

I think it's best to make returning Double.NaN normative (i.e. not allow flexibility for throwing an exception). The `NumberFormat.parse(String, ParsePosition)` methods return a failure value instead of throwing like `parse(String)` does. (E.g. DecimalFormat returns null on failed parse for 2 arg parse.)